post_office/mail.py

```diff
@@ -268,12 +278,28 @@ def send(email):
 
     connections.close()
 
-    # Update statuses of sent and failed emails
+    # Update statuses of sent emails
     email_ids = [email.id for email in sent_emails]
     Email.objects.filter(id__in=email_ids).update(status=STATUS.sent)
 
-    email_ids = [email.id for (email, e) in failed_emails]
-    Email.objects.filter(id__in=email_ids).update(status=STATUS.failed)
+    # Update statuses and conditionally requeue failed emails
+    num_failed, num_requeued = 0, 0
+    max_retries = get_max_retries()
+    scheduled_time = timezone.now() + get_time_delta_to_retry()
+    email_ids = [email.id for email, _ in failed_emails]
+    emails_failed = Email.objects.filter(id__in=email_ids)
```

Comment:
Do we need to re-fetch these emails from the DB? Can't we just use `failed_emails`?

Reply:
Yes indeed, `emails_failed` does not have to be a queryset, it can also be a list of objects.